### DISCUSSION TITLE
`import` (or `source`) in addition to `include` (alternative to "flattening")

### DISCUSSION THREAD

**Discussion by tomharrisonjr** [go-task/task#1670] [Ideas]
I just spent some significant effort reworking a few projects so that they can all share common tasks via `include`.  When I first read about `include` I assumed they were just a simple C-style import (or bash `source`).  But include is more sophisticated ... but that's not always what's needed.  Include creates an implicit namespace, and creates several other dependencies such as schema -- so that included tasks become more like callable functions. This is valid, and needed in many cases, but more work and planning.

In my case (and based on the recurring questions from others) there may be simpler use-cases.  For example, each of my projects has a `docker compose up -d` and a `docker compose down` task, along with a number of others that vary little or none between projects.  Some of the commands get tricky as you start adding options and environments, and others are pipelines of several commands, so mostly I am just insulating users from needing to care, and ensuring consistency.

In this case, an alternate command such as `import` that simply reads the file into the current context would seem to solve the simpler requirement.  The file would simply lists tasks that are suitable for import into a project as though they had been written there. (For the file to be valid YAML, perhaps just include the `tasks:` element, and ignore).

**Comment by trulede** [comment]
You might want to take a look at this [comment](https://github.com/go-task/task/issues/273#issuecomment-2142639593) where I suggest a way method which can be used to realise this use case. I encourage you to try it.

**Comment by andreynering** [comment]
Closing as a duplicate of #273 #1704.
